coderd/files/cache.go

```diff
@@ -105,11 +190,19 @@ func (c *Cache) Release(fileID uuid.UUID) {
 		return
 	}
 
+	c.currentOpenFileReferences.Dec()
 	entry.refCount--
 	if entry.refCount > 0 {
 		return
 	}
 
+	c.currentOpenFiles.Dec()
+
+	ev, err := entry.value.Load()
+	if err == nil {
+		c.currentCacheSize.Add(-1 * float64(ev.size))
+	}
```

Comment:
This feels really unfortunate, but it is all cached so it's not an expensive call.